specs/derivation.md

```diff
@@ -974,46 +979,171 @@ Some network upgrades require automated contract changes or deployments at speci
 To automate these, without adding persistent changes to the execution-layer,
 special transactions may be inserted as part of the derivation process.
 
-#### Eclipse: L1Block predeploy upgrade
+#### Ecotone
 
```

Comment:
Why does this need to be the block before the hardfork? If it contains no user transactions then it could just be the hardfork block and then hardfork block + 1 could call the new function on the `L1Block` contract

Reply:
Yeah, we just have to choose two blocks, either `(ecotone-1, ecotone)` or `(ecotone, ecotone+1)` to deploy, then switchover. 

My thinking was making the deploys happen in `ecotone-1`, means everything else can switch in `ecotone`. 

However I think `(ecotone, ecotone+1)` maybe a better choice as we can do:

* Network upgrades when parentBlock != ecotone && nextBlock == ecotone
* Everything else when parentBlock == ecotone && nextBlock == ecotone

I'll update this to specify ecotone.